### PR TITLE
Payout user offer

### PR DESCRIPTION
**Pull request by jdungan**
This refunds the users offer if they end up claiming a payout. This is for #341.

This credits a new PayoutCredit model which is based on the Fee meta class.

### CODE REVIEW

File: auctions/admin.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from django.contrib import admin
 
-from .models import Bid, Issue, Claim, Vote, Offer, Payout, OfferFee, PayoutFee
+from .models import Bid, Issue, Claim, Vote, Offer, Payout
+from .models import OfferFee, PayoutFee, PayoutCredit
```
Comment (groovecoder) [inline auctions/admin.py]: usual syntax for this is:
```
from .models import (Bid, Issue, Claim, Vote, Offer, Payout
                     OfferFee, PayoutFee, PayoutCredit)
```

File: auctions/models.py [hunk 2]
```diff
@@ -528,27 +529,51 @@ def __unicode__(self):
...
+                    payout=self,
+                    fee_type='refund',
+                    description="Your offer",
+                    amount=total_refund,
+                )
+                refund_user_offer.save()
+                total_payout_amount += total_refund
+
+        except Offer.DoesNotExist:
+            refund_user_offer = None
```
Comment (groovecoder) [inline auctions/models.py]: This will probably conflict with any other changes going into `Payout.request` ... merge this first?

Comment (jdungan) [inline auctions/models.py]: Yes.  I'll merge the new master with the other PR

**Comment by groovecoder** [comment]
Crap. Diving into this but I messed up my local postgres install at NDoCH trying to install GeoDjango & PostGIS. Will blow it all away and re-install ...

**Comment by groovecoder** [comment]
Now this needs its rebase.

**Comment by groovecoder** [comment]
Rebased and cleaned up the import statement comment I made. Do we want to include a comment or something about setting `api_success` to `True` when we do manual PayPal payments?

**Pull request merged**